src/MetaMorpho.sol

```diff
@@ -376,24 +375,31 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
             if (supplyCap == 0) revert ErrorsLib.UnauthorizedMarket(id);
 
-            (uint256 suppliedAssets,) =
+            (uint256 suppliedAssets, uint256 suppliedShares) =
                 MORPHO.supply(allocation.marketParams, allocation.assets, allocation.shares, address(this), hex"");
 
             if (_supplyBalance(allocation.marketParams) > supplyCap) {
                 revert ErrorsLib.SupplyCapExceeded(id);
             }
 
             totalSupplied += suppliedAssets;
+
+            emit EventsLib.ReallocateSupply(id, suppliedAssets, suppliedShares);
         }
 
+        uint256 newIdle;
         if (totalWithdrawn > totalSupplied) {
-            idle += totalWithdrawn - totalSupplied;
+            newIdle = idle + totalWithdrawn - totalSupplied;
         } else {
             uint256 idleSupplied = totalSupplied - totalWithdrawn;
             if (idle < idleSupplied) revert ErrorsLib.InsufficientIdle();
 
-            idle -= idleSupplied;
+            newIdle = idle - idleSupplied;
         }
+
+        idle = newIdle;
+
+        emit EventsLib.ReallocateIdle(newIdle);
```

Comment:
Do you think it could make sense to also track the `prevIdle` to easily calc the delta?